frontend/components/receipts/utils/order/pegged-order-info.tsx

```diff
@@ -0,0 +1,19 @@
+import { PriceWithTooltip } from '../string-amounts/price-with-tooltip.tsx'
+import { PeggedOrder } from '@vegaprotocol/types'
+
+interface PeggedOrderInfoProps {
+  peggedOrder: PeggedOrder
+  marketId: string
+}
+
+export const PeggedOrderInfo = ({ peggedOrder, marketId }: PeggedOrderInfoProps) => {
+  const { offset, reference } = peggedOrder
+  const formattedReference = reference.toLowerCase().replace('pegged_reference_', '').replace('_', ' ')
```

Comment:
We've used maps for this kind thing in all other places, I'd like to keep doing that so when the inevitable ask to translate the app comes we can use that.